### PR TITLE
Attach NgRestModel behaviors via attachBehaviors

### LINKED ISSUE #104: ngRest model behaviors breaks any DropDown Select
Hi!

Assume there are two ngRest models First and Second.

Second model has `first_id` attribute to use it as relation to First.

Adding ngRest plugin `selectModel` works just fine 

```php
/**
 * @inheritdoc
 */
public function ngRestAttributeTypes()
{
    return [
        'description' => 'text',
        'first_id' => [
            'selectModel',
            'modelClass' => First::className(),
            'valueField' => 'id',
            'labelField' => 'title'
        ],
    ];
}
```

unless any Second model behavior defined e.g.

```php
public function behaviors()
{
    return [
        'sluggable' => [
            'class' => SluggableBehavior::className(),
            'attribute' => 'description',
            'slugAttribute' => 'alias',
            'ensureUnique' => true,
        ],
        'timestamp' => [
            'class' => TimestampBehavior::className(),
            'attributes' => [
                ActiveRecord::EVENT_BEFORE_INSERT => ['created_at', 'updated_at'],
                ActiveRecord::EVENT_BEFORE_UPDATE => ['updated_at'],
            ],
        ],
    ];
}
```

Module to test to https://github.com/nick-denry/luya-behavior-bug-testmod

Install, test Second model crud and uncomment behavior:

https://github.com/nick-denry/luya-behavior-bug-testmod/blob/34d4d50b3259fb689b7e93950063fb8aaa99e2b4/models/Second.php#L70

The same thing for a SelectArray plugin.

### PR DESCRIPTION
**Pull request by nick-denry**
Close #104

### CODE REVIEW

**Comment by nick-denry** [comment]
>Could you please provide a very basic unit teste

Do you mean simple assertion with call of `behaviors()` or what?

File: src/ngrest/base/NgRestModel.php [hunk 1]
```diff
@@ -52,13 +52,13 @@
...
+     * Attach behaviors at constructor to prevent override them with call of behaviors() at subclass model.
      */
-    public function behaviors()
+    public function __construct($config = [])
     {
-        return [
+        $this->attachBehaviors([
             'NgRestEventBehvaior' => [
                 'class' => NgRestEventBehavior::className(),
                 'plugins' => $this->getNgRestConfig()->getPlugins(),
```
Comment (nick-denry) [inline src/ngrest/base/NgRestModel.php]: This line throws error

The scenario 'restcreate' does not exists in your scenarios list, have you forgote to defined the 'restcreate' in the scenarios() method?

**Comment by nadar** [comment]
1. Maye could use attach the behaviors before init method: https://github.com/yiisoft/yii2/blob/master/framework/db/BaseActiveRecord.php#L922-L926 and then call parent initializer.

2. Make a very basic unit tests, then you don't have to provide stack trace, we can see the problem on travis directly.

3. The scenario exception says that the ngrest plugin field is not inside the scenario this means you have no rule defined for the field.

**Comment by nadar** [comment]
@nick-denry do you need help on this?

**Comment by nadar** [comment]
@nick-denry A very basic test could look like this maybe:

Add new method `testBehaviorIsAttached` to https://github.com/luyadev/luya-module-admin/blob/master/tests/admin/ngrest/base/NgRestModelTest.php

```php
public function testBehaviorIsAttached()
{
    $model = new TestNgRestModel();
    $behaviors = $model->getBehaviors();

    // now check if the behaviors has been attached to the $model $this->assertArrayHasKey(...) for example
}
```

**Comment by nick-denry** [comment]
Ahh this way. Fine. I'll make a test within next 24h.

Many thanks, @nadar

**Comment by nadar** [comment]
@nick-denry perfect. I also think as mentioned (https://github.com/luyadev/luya-module-admin/pull/106#issuecomment-383817538) maybe we should move to init() method, but not sure when the behaviors are attached and the after records events are triggered.

**Comment by nadar** [comment]
@nick-denry i still think your improvement is a great idea. Any news on this or should we close the PR?

**Comment by nadar** [comment]
@nick-denry i did the implementation including unit tests and changelog.

**Pull request closed without merging**